client/src/components/Homepage/Homepage.js

```diff
@@ -6,39 +6,46 @@ import About from "../About/About";
 import Footer from "../Footer/Footer";
 import { decode } from "jsonwebtoken";
 import { useState, useEffect } from "react";
+import { refresh } from '../../api/index';
 
 const Homepage = () => {
   const navigate = useNavigate();
   const user = JSON.parse(localStorage.getItem("profile"));
   const [userRole, setUserRole] = useState("");
 
-  if (!user) {
-    navigate("/login");
-  }
-
   useEffect(() => {
-    const checkUserRole = async () => {
+    let timeoutId;
+
+    const refreshToken = async () => {
       try {
-        const decodedToken = decode(user.token);
-        if (decodedToken) {
-          const userRole = decodedToken.role;
-          setUserRole(userRole);
-        } else {
-          setUserRole("");
+        const response = await refresh({ token: user?.refreshToken });
+        if (response.statusText !== 'OK') {
+          localStorage.removeItem("profile");
+          navigate("/login");
         }
+        const newToken = response.data.token;
+        setUserRole(decode(newToken)?.role)
+        localStorage.setItem("profile", JSON.stringify({ ...user, token: newToken }));
       } catch (error) {
-        console.error("Error fetching user profile:", error);
-        setUserRole("");
+        console.log(error)
+        localStorage.removeItem("profile");
+        setUserRole("")
+        navigate("/login");
+      } finally {
+        timeoutId = setTimeout(refreshToken, 8000);
       }
     };
```

Comment:
It is taking tim efor the "Get interviewed by AI" text on homepage to load 
![image](https://github.com/rasikaghadge/AI-Automated-Interview-Platform/assets/90198493/68a8cde4-7a72-4a68-bb41-cb38234dd30b)